### DISCUSSION TITLE
Rich result panel like Spotlight

### DISCUSSION THREAD

**Discussion by LeLocTai** [Flow-Launcher/Flow.Launcher#1357] [Ideas]
A side panel that can display arbitrary content like used in Spotlight is very useful, especially when combined with plugins.

![image](https://user-images.githubusercontent.com/12182348/187934535-ee954ff5-e8eb-441a-bab9-1dd5a33ab05b.png)

**Comment by onesounds** [comment]
<img src="https://user-images.githubusercontent.com/6903107/192424643-8bbc9654-e76c-419f-93f2-bbdbc42228d7.png" width="500">

- A preview feature is being prepared. (maybe next major version?)
- Preview from Image and high res video thumb and full file path

<img src="https://user-images.githubusercontent.com/6903107/192424765-71cbd021-8033-493b-9b04-2b9190ba9452.png" width="400">

- We are testing the search results in preview panel, but it will take time because we think it need to render the contents separately.

**Comment by LeLocTai** [comment]
One of the big reason that I chose flow is that it is fast. Using web view I believe is a step in the wrong direction in that regard.
I was thinking something more of a plugin provided panel, like the current setting panel.

**Reply by onesounds** [reply]
> One of the big reason that I chose flow is that it is fast. Using web view I believe is a step in the wrong direction in that regard. I was thinking something more of a plugin provided panel, like the current setting panel.

That's how it's being produced now. As mentioned above, We do not intend to include webviews.
We are studying ways to make the search results look clean. (not easy in wpf 😂)

**Reply by taooceros** [reply]
Is speed really mattering for previewing panel? The webview will only be used when previewing (and i think we can cache it).

**Reply by onesounds** [reply]
> Is speed really mattering for previewing panel? The webview will only be used when previewing (and i think we can cache it). 

actually, speed is 2nd matter. design issue is 1st. I researched link preview(aka url preview. garulf mentioned it.) function. I think we should use this. (but there a not good solution in wpf. 😂)

**Reply by Garulf** [reply]
In my opinion we should definitely not show a web view. It  doesn't fit the design very well. 

I believe a web snippet like what discord shows for a URL would work best. 

![image](https://user-images.githubusercontent.com/535299/192566979-d9a7c43d-df15-412f-80eb-f731d281850a.jpeg)

Ideally with a larger image preview.
